src/test/java/com/lgtm/easymoney/controllers/SearchControllerTest.java

```diff
@@ -0,0 +1,189 @@
+                jsonPath("$.userProfiles[0].email").value(email),
+                jsonPath("$.userProfiles[0].phone").value(phone),
+                jsonPath("$.userProfiles[0].userType").value(String.valueOf(userType)));
+    }
+
+    @Test
+    public void searchByNameSuccess() throws Exception {
+    // Arrange
+    Mockito.when(searchService.searchByInfo(accountName))
+        .thenReturn(ResponseEntity.of(Optional.of(searchRsp)));
+
+    // Act
+    ResultActions returnedResponse = getSearchByInfo(accountName);
+
+    // Assert
+    returnedResponse.andExpectAll(
+        status().isOk(),
+        jsonPath("$.success").value(true),
+        jsonPath("$.userProfiles[0].uid").value(uid),
+        jsonPath("$.userProfiles[0].accountName").value(accountName),
+        jsonPath("$.userProfiles[0].address").value(address),
+        jsonPath("$.userProfiles[0].email").value(email),
+        jsonPath("$.userProfiles[0].phone").value(phone),
+        jsonPath("$.userProfiles[0].userType").value(String.valueOf(userType)));
+    }
+
+    @Test
+    public void searchByEmailSuccess() throws Exception {
+        // Arrange
+        Mockito.when(searchService.searchByInfo(email))
+                .thenReturn(ResponseEntity.of(Optional.of(searchRsp)));
+
+        // Act
+        ResultActions returnedResponse = getSearchByInfo(email);
+
+        // Assert
+        returnedResponse.andExpectAll(
+                status().isOk(),
+                jsonPath("$.success").value(true),
+                jsonPath("$.userProfiles[0].uid").value(uid),
+                jsonPath("$.userProfiles[0].accountName").value(accountName),
+                jsonPath("$.userProfiles[0].address").value(address),
+                jsonPath("$.userProfiles[0].email").value(email),
+                jsonPath("$.userProfiles[0].phone").value(phone),
+                jsonPath("$.userProfiles[0].userType").value(String.valueOf(userType)));
+    }
+
+    @Test
+    public void searchByPhoneSuccess() throws Exception {
+        // Arrange
+        Mockito.when(searchService.searchByInfo(phone))
+                .thenReturn(ResponseEntity.of(Optional.of(searchRsp)));
+
+        // Act
+        ResultActions returnedResponse = getSearchByInfo(phone);
+
+        // Assert
+        returnedResponse.andExpectAll(
+                status().isOk(),
+                jsonPath("$.success").value(true),
+                jsonPath("$.userProfiles[0].uid").value(uid),
+                jsonPath("$.userProfiles[0].accountName").value(accountName),
+                jsonPath("$.userProfiles[0].address").value(address),
+                jsonPath("$.userProfiles[0].email").value(email),
+                jsonPath("$.userProfiles[0].phone").value(phone),
+                jsonPath("$.userProfiles[0].userType").value(String.valueOf(userType)));
+    }
+
+    @Test
+    public void searchByIdFailedWithInvalidPathVar() throws Exception {
+        Mockito.doThrow(new NumberFormatException(null))
```

Comment:
where's the act and assert?

Reply:
Example: https://github.com/howieraem/COMS4156-ASE/blob/e4035f78c14ebfe3f8a738f1fd8f216f3113c563/src/test/java/com/lgtm/easymoney/controllers/GroupControllerTest.java#L228